src/amplitude-client.js

```diff
@@ -556,32 +556,36 @@ AmplitudeClient.prototype._unsentCount = function _unsentCount() {
 };
 
 /**
- * Send events if ready. Returns true if events are sent.
- * @private
+ * Flushes batched events if ready. Is periodically called by SDK,
+ * but can be manually called by client. Behavior is dependent on configuration
+ * If `batchEvents === false`, it is called after `logEvent`.
+ * If `batchEvents === true`, then events are sent only when batch criterias are met.
+ * @public
+ * @returns {boolean} true if events are sent. false if they aren't or are scheduled for a later time
```

Comment:
if the API key is not set, this function will fail at `this._sendEvents()` on L1524 and return true. Should we account for that behavior?

Reply:
[I added a check](https://github.com/amplitude/Amplitude-JavaScript/blob/feat-flush-events/src/amplitude-client.js#L571-L574)